airflow/api_fastapi/core_api/routes/public/dags.py

```diff
@@ -303,3 +307,45 @@ def delete_dag(
             status.HTTP_409_CONFLICT, f"Task instances of dag with id: '{dag_id}' are still running"
         )
     return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@dags_router.get(
+    "/{dag_id}/assets/queuedEvent",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_401_UNAUTHORIZED,
+            status.HTTP_403_FORBIDDEN,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+)
+def get_dag_asset_queued_events(
+    dag_id: str,
+    session: Annotated[Session, Depends(get_session)],
+    before: str = Query(None),
+) -> QueuedEventCollectionResponse:
+    """Get queued asset events for a DAG."""
+    where_clause = [AssetDagRunQueue.target_dag_id == dag_id]
+    if before:
+        before_parsed = timezone.parse(before)
+        where_clause.append(AssetDagRunQueue.created_at < before_parsed)
+    query = (
+        select(AssetDagRunQueue, AssetModel.uri)
+        .join(AssetModel, AssetDagRunQueue.asset_id == AssetModel.id)
+        .where(*where_clause)
+    )
+    result = session.execute(query).all()
```

Comment:
You can also easily add the pagination, add we do for other endpoints, really easy to do with our utility code. This will prevent you from having to do manually:
```
result = session.execute(query).all()
total_entries = len(result)

```

Reply:
Yeah let me try that as well. The reason I didnt use that mainly was because there's no real pagination for this.
limit, order_by etc dont exist in this context but I understand that I can set those to None and make do.

Should we consider renaming the `paginated_select` function?

Reply:
cc @pierrejeambrun just adding you for above comment but I have handled it already though paginated_select in the latest commit